homeassistant/components/sensor/sma.py

```diff
@@ -95,11 +95,14 @@ def _check_sensor_schema(conf):
         config_sensors = {s.name: [] for s in sensor_def}
 
     # Prepare all HASS sensor entities
+
+    smaid = ''.join(['{:02x}'.format(int(v))
+                     for v in config[CONF_HOST].split('.')])
```

Comment:
That assumes that host provided is an IP address, and thus fails if a hostname is provided instead in the configuration. Why not just use CONF_HOST itself as the part of the id?

Reply:
We can't use ip address or hostname in unique id, since those are not stable.